src/game/enemy/components.rs

```diff
@@ -1 +1,40 @@
-use bevy::prelude::*;
\ No newline at end of file
+use bevy::prelude::*;
+use std::fmt::{Display, Formatter, Result};
+
+#[derive(Component)]
+pub struct EnemyWord {
+    pub cur: usize, // Pointer to current letter for word match
+    pub word: String,
+}
+
+impl Default for EnemyWord {
+    fn default() -> Self {
+        EnemyWord {
+            cur: 0,
+            word: "**default**".into(),
+        }
+    }
+}
+
+impl Display for EnemyWord {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        write!(f, "{}", self.word)
+    }
+}
+
+#[derive(Component)]
+pub struct Enemy {
+    pub health: usize, // Percentage [0, 100]
+    pub direction: Vec2,
+    pub word: EnemyWord,
+}
+
+impl Default for Enemy {
+    fn default() -> Self {
+        Enemy {
+            health: 100,
+            direction: Vec2::default(),
```

Comment:
Whats the idea behind direction?

Reply:
In the tutorial I did they used this to move the sprite every update. It gives really smooth movement, but the more I think of it we may just want to store a point on the map that it will gradually move towards.
```
pub fn enemy_movement(mut enemy_query: Query<(&mut Transform, &Enemy)>, time: Res<Time>) {
    for (mut transform, enemy) in enemy_query.iter_mut() {
        let direction = Vec3::new(enemy.direction.x, enemy.direction.y, 0.0);
        transform.translation += direction * ENEMY_SPEED * time.delta_seconds();
    }
}
```